### PR TITLE
Feature/automatic spell checking

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/nextcloud/neon/issues/834

### CODE REVIEW

**Comment by provokateurin** [comment]
Works as expected: https://github.com/nextcloud/neon/actions/runs/6284283715/job/17065500854

**Review by Leptopoda — COMMENTED** [review]
I think we should reorganize the dictionaries into:
- dart/flutter (only containing dart stuff)
- dependencies/libraries (containing stuff we can't controll like rxdart or zxing
- misc
- custom

For some less common things you should also add some comments like what even is `seti` or  `siacs`

File: .cspell/dart.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+autofocus
+datetime
```
[the remark this answers is not in the document]
Comment (provokateurin) [reply .cspell/dart.txt:2]: Ah true, I thought this was the DateTime class

**Comment by provokateurin** [comment]
I added a script to update the dictionaries (to remove outdated words).
I wasn't sure how to best organize the dictionaries, but we can try your idea.

**Comment by provokateurin** [comment]
I'm not sure why the CI has a problem with `openapi` while I locally don't have the problem (running the latest version).

**Review by Leptopoda — COMMENTED** [review]
But now we have some duplications like `foss` and `FOSS` and it was passing fine without them.

**Comment by provokateurin** [comment]
This depends on https://github.com/nextcloud/notifications/pull/1673

**Review by Leptopoda — APPROVED** [review]

**Pull request merged**